webapp/channels/src/components/admin_console/data_retention_settings/global_policy_form/global_policy_form.tsx

```diff
@@ -135,22 +137,52 @@ export default class GlobalPolicyForm extends React.PureComponent<Props, State>
         return true;
     };
 
-    setRetentionHours = (dropdownValue: string, value: string): number => {
+    setRetentionDays = (dropdownValue: string, value: string): number => {
         if (dropdownValue === YEARS) {
-            return parseInt(value, 10) * 24 * 365;
+            return parseInt(value, 10) * 365;
         }
 
         if (dropdownValue === DAYS) {
-            return parseInt(value, 10) * 24;
+            return parseInt(value, 10);
         }
 
+        return 0;
+    };
+
+    setRetentionHours = (dropdownValue: string, value: string): number => {
         if (dropdownValue === HOURS) {
             return parseInt(value, 10);
         }
 
         return 0;
     };
 
+    getConfigValue(config: DeepPartial<AdminConfig> | Partial<EnvironmentConfig>, path: string) {
+        const pathParts = path.split('.');
+
+        return pathParts.reduce((obj: object | null, pathPart) => {
+            if (!obj) {
+                return null;
+            }
+
+            return obj[pathPart as keyof object];
+        }, config);
+    }
+
+    isSetByEnv = (path: string) => {
+        return Boolean(this.getConfigValue(this.props.environmentConfig, path));
+    };
+
+    isMessageRetentionSetByEnv = () => {
```

Comment:
I think it would be simpler to simply remove the above two functions and just use the `environmentConfig` object directly?
Like:
```js
isMessageRetentionSetByEnv = () => {
        return (this.props.environmentConfig.DataRetentionSettings?.MessageRetentionDays && ...restOfTheCode);
    };
```

Same for the file func.

Reply:
Good point, fixed